app/src/lib/stores/helpers/infer-compare-to-branch.ts

```diff
@@ -0,0 +1,153 @@
+import { Branch, IAheadBehind } from '../../../models/branch'
+import { PullRequest } from '../../../models/pull-request'
+import { getAheadBehind, revRange } from '../../git'
+import { Repository } from '../../../models/repository'
+import { GitHubRepository } from '../../../models/github-repository'
+
+interface IOfGithub {
+  kind: 'github'
+  repository: Repository
+}
+
+interface IOfPullRequest {
+  kind: 'pr'
+  pullRequest: PullRequest
+}
+
+interface IOfFork {
+  kind: 'fork'
+  repository: Repository
+  gitHubRepository: GitHubRepository
+  currentBranch: Branch
+}
+
+// I am struggling to come up with a decent name for this type
+type NamingIsHard = IOfPullRequest | IOfGithub | IOfFork
+
+/**
+ * Infers which branch to use as the comparison branch
+ *
+ * @param branches The list of all branches for the repository
+ * @param needsAName The object that determines how the branch will be inferred
+ */
+export async function inferComparisonBranch(
+  branches: ReadonlyArray<Branch>,
+  needsAName: NamingIsHard
+): Promise<Branch | null> {
+  switch (needsAName.kind) {
```

Comment:
Hopefully the fact that this function is just a switch statement eliminates the need to write tests to exercise it. I have a few stashed away, but they are literally duplicates of the test cases for each of the _private_ functions.